libfwupdplugin/fu-sbatlevel-section-firmware.c

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2023 Canonical Ltd.
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#define G_LOG_DOMAIN "FuFirmware"
+
+#include "config.h"
+
+#include "fu-bytes.h"
+#include "fu-mem.h"
+#include "fu-sbatlevel-section-firmware.h"
+#include "fu-sbatlevel-section-struct.h"
+
+G_DEFINE_TYPE(FuSbatlevelSectionFirmware, fu_sbatlevel_section_firmware, FU_TYPE_FIRMWARE);
+
+static gboolean
+fu_sbatlevel_section_add_entry(FuFirmware *firmware,
+			       GBytes *fw,
+			       const guint8 *buf,
+			       gsize bufsz,
+			       gsize entry_offset,
+			       const gchar *entry_name,
+			       guint64 entry_idx,
+			       FwupdInstallFlags flags,
+			       GError **error)
+{
+	g_autoptr(FuFirmware) entry_fw = NULL;
+	g_autoptr(GBytes) entry_blob = NULL;
+	gsize entry_size;
+
+	for (entry_size = 0; ((entry_offset + entry_size) < bufsz); ++entry_size) {
+		if (buf[entry_offset + entry_size] == 0)
+			break;
+	}
```

Comment:
i don't really get what this is doing

Reply:
Looking for the null character terminator.